### DISCUSSION TITLE
convert eforms xml to old text format

### DISCUSSION THREAD

**Discussion by hornschorsch** [OP-TED/eForms-SDK#757] [Q&A] [answered]
Hello,

does anyone have a software that can convert the eforms xml files into the old text format?

Or is there a list that states which element of the xml files equates to which part of the old text format?

For example in the TED-xml-files this was as follows:

TI (Title) is /TED_EXPORT/TRANSLATION_SECTION/ML_TITLES/ML_TI_DOC[@LG="DE"]/*//text()
PD (Publication Date) is /TED_EXPORT/CODED_DATA_SECTION/REF_OJS/DATE_PUB/text()
ND (Number of Document) is /TED_EXPORT/CODED_DATA_SECTION/NOTICE_DATA/NO_DOC_OJS/text()

etc.

Background is, i have an old application that reads the old text files that where published on the TED ftp server. A few years ago, when the text files where replaced by the ted xml files, i wrote a small converter that converts the xml to text which then can be read by my application. I now try to do the same with the eforms xml.

**Comment by pdonohoe** [comment, marked as answer]
The Publications Office has no software to convert eforms XML files to the old TED XML format. The closest thing is the TED XML Data Converter which does the opposite: it converts notices in TED XML into partial eForms XML notices. The source code for this converter is available at: https://github.com/OP-TED/ted-xml-data-converter and there is an API for it: https://enotices2.preview.ted.europa.eu/esenders/webjars/swagger-ui/index.html#/Notice/convert
https://enotices2.ted.europa.eu/esenders/webjars/swagger-ui/index.html#/Notice/convert

Please be aware that as the amount of information which can be reported in eForms notices is greater than was possible in the TED notices, some information included in eForms has no equivalent in TED, and would not be able to be converted to the TED format.

**Comment by hornschorsch** [comment]
Thanks for your reply! We rewrote our application so that it converts the eforms xml files into html an reads these into the database. Now we have the problem that in the daily bulk download of xml files there are still a lot of old TED XML files.

If i try to convert them to eforms XML with:

curl 'https://enotices2.ted.europa.eu/esenders/api/v2/notice/notices/convert' -H 'Content-Type: application/xml' -d @xmlfile

i get a 400 Missing Authorization header in request.

In the original version of your answer you mentioned another API:

curl -s 'https://converter.ted.europa.eu/api/v1/convert' -H 'Content-Type: application/xml' -d @xmlfile

which one is the correct one (i believe the first) and how do i get a username/password for it?

Can this API be used unlimited or should i install the ted-xml-data-converter locally?
